WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -362,6 +366,48 @@ - (ReaderTopic *)siteTopicForPost:(ReaderPost *)post
     return topic;
 }
 
+- (void)siteTopicForSiteWithID:(NSNumber *)siteID
+                       success:(void (^)(NSManagedObjectID *objectID, BOOL isFollowing))success
+                       failure:(void (^)(NSError *error))failure
+{
+    ReaderTopicServiceRemote *remoteService = [[ReaderTopicServiceRemote alloc] initWithApi:[self apiForRequest]];
+    [remoteService fetchSiteInfoForSiteWithID:siteID success:^(RemoteReaderSiteInfo *siteInfo) {
+        if (!success) {
+            return;
+        }
+        NSEntityDescription *entity = [NSEntityDescription entityForName:NSStringFromClass([ReaderTopic class])
+                                                  inManagedObjectContext:self.managedObjectContext];
+        ReaderTopic *topic = [[ReaderTopic alloc] initWithEntity:entity
+                     insertIntoManagedObjectContext:self.managedObjectContext];
```

Comment:
With the current implementation, it will only ever be the main context so the perform block is just extra work.  Since you mentioned notifications, what are your thoughts there?  Main context or derived.  I can the perform block back if there's a need.